.github/codeql/queries/assert-pure.ql

```diff
@@ -1,21 +1,44 @@
 /**
  * @name Unwanted dependency on vscode API
- * @kind problem
+ * @kind path-problem
  * @problem.severity error
  * @id vscode-codeql/assert-pure
  * @description The modules stored under `pure` and tested in the `pure-tests`
  * are intended to be "pure".
  */
+
 import javascript
 
-class VSCodeImport extends ASTNode {
-  VSCodeImport() {
-    this.(Import).getImportedPath().getValue() = "vscode"
+class VSCodeImport extends AstNode {
+  VSCodeImport() { this.(Import).getImportedPath().getValue() = "vscode" }
+}
+
+class PureFile extends File {
+  PureFile() {
+    (
+      this.getRelativePath().regexpMatch(".*/src/pure/.*") or
+      this.getRelativePath().regexpMatch(".*/src/common/.*")
+    ) and
+    not this.getRelativePath().regexpMatch(".*/src/common/vscode/.*")
   }
 }
 
+Import getANonTypeOnlyImport(Module m) {
+  result = m.getAnImport() and not result.(ImportDeclaration).isTypeOnly()
+}
+
+Module getANonTypeOnlyImportedModule(Module m) {
+  result = getANonTypeOnlyImport(m).getImportedModule()
+}
+
+query predicate edges(AstNode a, AstNode b) {
```

Comment:
Does there need to be a `nodes` query? I'm not 100% sure how this works.

Reply:
I guess it's working based on your screenshot. Have you already fixed the alerts? Otherwise, I'm not sure why code scanning isn't reporting them.

Reply:
If `nodes` isn't provided it defaults to all possible values of the columns in `edges`, so it's ok unless you need to constraint that set further.

Reply:
They do show up in code-scanning for me. They won't show up on the PR because the alerts don't appear in the diff, but you can see them on the security tab:
![Screenshot 2023-05-30 at 11 32 43](https://github.com/github/vscode-codeql/assets/3749000/03d24d46-ce17-4b44-915b-a8980e8c76a0)

And comparing to main:
![Screenshot 2023-05-30 at 11 32 31](https://github.com/github/vscode-codeql/assets/3749000/0e005384-12b5-4018-a4fb-6476cc149815)